### ISSUE TITLE
@level selector

### ISSUE THREAD

**Issue by MartinPacker** [MartinPacker/filterCSV#30]
For example `@1` to select root nodes, `@1` to select Level 1 nodes. And so on.

**Comment by EkanshBhatnagar** [comment]
Want to contribute!! Can I get some details about the issue?

**Comment by MartinPacker** [comment]
Hello Ekansh! I'll explain the issue in more detail below but I first have to ask if you're using filterCSV already - and how you're finding it. And what you do with the output.

So, a mindmap is generally a tree structure. The root of the tree is generally at Level 0. (In fact our data structures have -1 as the root level because there might be several trees in the forest and iThoughts supports them as Level 0 nodes. The children of Level 0 nodes are Level 1 - and so on.

The objective of this enhancement could be to take, say, all the Level 1 nodes and make them Level 0, placing them just after their parent nodes. Or make them replace the parent - with `promote`. Or it could be to take the Level 2 nodes and make them yellow, or a triangle, or something.

So the resolution to this issue would be to find an easy way of saying "select all Level n nodes".

Basically it's walking the tree, picking up nodes.

**Comment by EkanshBhatnagar** [comment]
I haven't used filterCSV yet. I though have used iThoughts and found it quite useful. I am a intermediate programmer, and would be very willing to try solving this issue, as it will give me a opportunity to learn.
Can you describe and help me where to start? I know the data structures well and can help.

**Comment by MartinPacker** [comment]
I'm not trying to put you off but I'd suggest you try out filterCSV - just matching some nodes with regexes and setting their colours:

1. Export a simple sample from iThoughtsX as CSV.
2. Throw it through filterCSV, playing with filtering and colouring.
3. Import back into iThoughts.

That helps you understand what the result of a "Level 1" filter would be. Also gives you a chance to tell me how bad the documentation is. (I already know I need to add some graphics.)

**Comment by MartinPacker** [comment]
Shipped in 1.6. Closing
